Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+        {
+            float totalEvalDurations = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                UpdateEvalProgressUI(i);
+                float evalDuration = await System.Threading.Tasks.Task.Run(() => Evaluate(nonBatchingSession_, inputImages));
+                totalEvalDurations += evalDuration;
+            }
+            avgNonBatchedDuration = totalEvalDurations / 100;
+        }
+
+        private static float Evaluate(LearningModelSession session, List<VideoFrame> input)
+        {
+            string inputName = session.Model.InputFeatures[0].Name;
+            float totalDuration = 0;
+            for (int j = 0; j < input.Count; j++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                var binding = new LearningModelBinding(session);
+                var start = HighResolutionClock.UtcNow();
+                binding.Bind(inputName, input[j]);
+                session.Evaluate(binding, "");
+                var stop = HighResolutionClock.UtcNow();
+                var duration = HighResolutionClock.DurationInMs(start, stop);
+                totalDuration += duration;
+            }
+            return totalDuration;
+        }
+
+        async private System.Threading.Tasks.Task ClassifyBatched(List<VideoFrame> inputImages, int batchSize)
+        {
+            float totalEvalDurations = 0;
+            for (int i = 0; i < 100; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                UpdateEvalProgressUI(i);
+                float evalDuration = await System.Threading.Tasks.Task.Run(() => EvaluateBatched(BatchingSession_, inputImages, batchSize));
+                totalEvalDurations += evalDuration;
+            }
+            avgBatchDuration = totalEvalDurations / 100;
+        }
+
+        private static float EvaluateBatched(LearningModelSession session, List<VideoFrame> input, int batchSize)
+        {
+            // Evaluate
+            int numBatches = input.Count / batchSize;
+            string inputName = session.Model.InputFeatures[0].Name;
+            float totalDuration = 0;
+            for (int i = 0; i < numBatches; i++)
+            {
+                if (navigatingAwayFromPage)
+                    break;
+                var binding = new LearningModelBinding(session);
+                List<VideoFrame> batch = input.GetRange(batchSize * i, batchSize);
+                var start = HighResolutionClock.UtcNow();
+                binding.Bind(inputName, batch);
+                session.Evaluate(binding, "");
+                var stop = HighResolutionClock.UtcNow();
+                var duration = HighResolutionClock.DurationInMs(start, stop);
+                totalDuration += duration;
+            }
+            return totalDuration;
+        }
+
+        private int GetBatchSizeFromBatchSizeComboBox()
```

Comment:
We had talked about doing a slider or some way for the user to pick exactly the value they wanted. Why is it limited to these 4 choices?

Reply:
Since there are 50 inputs I decided to limit the choices for batch size to the numbers that 50 is evenly divisible by, otherwise there would be at least 1 uneven batch (batch with a different number of inputs than the rest). When trying to pass a different number of inputs than the session is created for an error would be thrown - so the only way to handle this is to create an entirely new session to handle the 1 batch of a different size. This would pretty drastically increase the amount of time for batching which wouldn't help showcase that batching inputs is faster than not batching inputs.